odoo/addons/base/models/ir_rule.py

```diff
@@ -143,6 +143,8 @@ def _compute_domain(self, model_name: str, mode: str = "read") -> Domain:
         # add rules for parent models
         global_domains: list[Domain] = []
         for parent_model_name, parent_field_name in model._inherits.items():
+            if not model._fields[parent_field_name].store:
+                continue
```

Comment:
That shortcut makes sense from a technical point of view, but it contradicts the current practice for inherited models: a record is accessible if it satisfies its model's access rules **and** its parent record is accessible, too.  On some models, the security rules are mostly based on those of the parent model.  The shortcut enables to access a record _without taking into account_ its parent record...

I am not against this change, but it kind of introduces a special case which might look like an access rights leak, and which developers must be aware of.

Reply:
It is really strange that we have different security based on fact that the field is stored or not.